### PR TITLE
Feature/implement logic - Milestone5

### PR DESCRIPTION
**Pull request by shubham14p3**
## Requirements

1. **App** component
    - The **App** component is going to be your stateful parent component. Initialize its state in the constructor with three
     properties: `total`, `next` and `operation`, just like in your `calculate.js` module. By default, they should all
     be set to `null`.
    - Implement the `handleClick` function which takes `buttonName` as an argument and changes the state based on the result of the `calculate` function.
    - Pass the `total` or `next` value to **Display** to display the current result.
    - Pass `handleClick` as a `clickHandler` property to **ButtonPanel**
2. **ButtonPanel**
    - Implement the `handleClick` function which takes `buttonName` as an argument and returns `clickHandler` from props.
    - Pass the `clickHandler` property to each button in the panel.
3. **Button**
    - Implement the `handleClick` function which takes `buttonName` as an argument and returns `clickHandler` from props.
    - Use the `onClick` event to attach the `handleClick` event to the button.
    - Documentation: https://reactjs.org/docs/events.html#mouse-events

Voila! You have finished your first React project. 
**General requirements:**

- Add a descriptive README file to your repositories as described.
- Use meaningful names for your Pull Requests and add a summary of its content as described.
- Use descriptive commit messages as described.
- Do not commit files that are not related to the project. Instead, add them to .gitignore file.
- Do not commit inline comments. They will make your project look messy. If you need to review a previous version of your code, you can always use git history.

**Git Flow:**

- Use a specific branch for each feature.
- Use a descriptive name for your branches.
- Use Rubocop to validate your code.

### CODE REVIEW

**Comment by shubham14p3** [comment]
Note : Heroku & Readme Files will be updated once the project get approved,
Happy Coding.

**Review by Oghenebrume50 — CHANGES_REQUESTED** [review]
### CHANGES REQUIRED

Hi @shubham14p3 good job working on this, everything works fine :+1:  just some few things to point out. 

- When a calculation is done and the result is being displayed, if a number is pressed, you want to replace the currently displayed answer with the number that was pressed, for example. 
Press 9 + 5, 14 will show on the screen, if you then press 3 immediately, the display should show only 3 and not 143.

- The % operator is not working, this should just return the num1 / 100 

Please make changes and request for another review :slightly_smiling_face: 

**RAPHAEL NORIODE** :mask: stay_safe

**Comment by shubham14p3** [comment]
Changes Applied as per requested

**Review by fatymahmed — CHANGES_REQUESTED** [review]
## Hi @shubham14p3 
You have done a great job on this milestone 👏👏
- The `%` operator is not working, kindly fix it.

## Status: Changes Required ♻️
_Feel free to reach out to me on slack (Fatima) if you have any questions or need any help_

**Comment by shubham14p3** [comment]
Thanks for the review.

Code was broken, I have fixed the '%' now every function is working fine, kindly check and verify.

**Review by fatymahmed — CHANGES_REQUESTED** [review]
## Hi @shubham14p3 
You have done a great job on this milestone 👏👏
- The `%` operator is not working, kindly fix it.
**See my comment below on how to fix this**
## Status: Changes Required ♻️
_Feel free to reach out to me on slack (Fatima) if you have any questions or need any help_

File: src/logic/operate.js [hunk 1]
```diff
@@ -1,30 +1,26 @@
...
+      break;
+    case '-': num = first.sub(second).toString();
+      break;
+    case '*': num = first.mul(second).toString();
+      break;
+    case '/':
+      num = numberTwo === '0' ? '0' : first.div(second).toString();
+      break;
+    case '%': num = first.div(second).toString();
+      break;
```
Comment (fatymahmed) [inline src/logic/operate.js]: - The logic doesn't seem right here, remember it should be `first/100`
Screenshot

**Review by fatymahmed — CHANGES_REQUESTED** [review]
## Hi @shubham14p3 
- The `%` operator is not working, kindly fix it.
**See my comment below on how to fix this**
## Status: Changes Required ♻️
_Feel free to reach out to me on slack (Fatima) if you have any questions or need any help_

File: src/logic/operate.js [hunk 2]
```diff
@@ -1,30 +1,26 @@
...
+    case '-': num = first.sub(second).toString();
+      break;
+    case '*': num = first.mul(second).toString();
+      break;
+    case '/':
+      num = numberTwo === '0' ? '0' : first.div(second).toString();
+      break;
+    case '%': num = first.mod(second).toString();
+      break;
```
Comment (fatymahmed) [inline src/logic/operate.js]: - This should not be mod, it should convert `first` to a percentage based number.  
`95%` should be `0.95`
Screenshot

**Comment by shubham14p3** [comment]
Finally Made the  changes as requested. 
 Happy Coding

**Comment by shubham14p3** [comment]
Note : My git Bash is broken , I am fixing it from Powershell , and everything is looking fine in local (No Error), but don't know why it is failing in the linter. Kindly Let it go , regards.

File: .eslintrc.json [hunk 3]
```diff
@@ -4,22 +4,27 @@
...
+	"arrow-parens": ["error", "as-needed"],
+	"react/prop-types":[0],
+	"react/require-default-props":[0],
+	"react/default-props-match-prop-types":[0],
+	"react/no-access-state-in-setstate":[0],
+	"react/button-has-type":[0]
   },
   "ignorePatterns": ["dist/", "build/"]
-}
+}
```
Comment (Torres-ssf) [inline .eslintrc.json]: - [x] Your eslintrc file is different from the one provided by Microverse. We should not change this file. We need to switch this file back to its original configuration, you can find it [here](https://github.com/microverseinc/linters-config/blob/master/react-redux/.eslintrc.json).

Check here how you can make your code in accordance if these rules:

- [react/prop-types](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prop-types.md)

- [react/require-default-props](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/require-default-props.md)

- [react/default-props-match-prop-types ](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/default-props-match-prop-types.md)

- [react/no-access-state-in-setstate](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-access-state-in-setstate.md)

- [react/button-has-type](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/button-has-type.md)

You can always try to google any rule that may cause trouble to you. I most cases is really simple to make your code comply with the rule.

**Review by Torres-ssf — CHANGES_REQUESTED** [review]
### Changes required :muscle: 
Hi @shubham14p3 :hand: 
First of all, great job on this milestone :clap: I can see that you have been working hard on it. Nice job :smile:  :+1: .
But we are not ready to move on yet. We just need to apply some changes first.

- [x] This linter.yml file is from the javascript module. We need to change it to the React module instead. You can find it [here](https://github.com/microverseinc/linters-config/blob/master/react-redux/.github/workflows/linters.yml).
https://github.com/shubham14p3/react-calculator/blob/2f8b349068de43e6cd4e5b61328a7ee936c7cb19/.github/workflows/linters.yml#L17-L22

- [x] For some reason the times operation it's not working properly right now. It always results in 0
bug_calculator1

Please kindly submit for code review again after making these changes.
Feel free to message me on slack if you have any questions :smile: 
Happy Coding :computer: 
**Sergio Torres**

File: .eslintrc.json (hunk 3 above)
Comment (shubham14p3) [reply .eslintrc.json]: Have resolved everything, need to update entire logic, Kindly check the working ,eslint & do let me know if everything looks fine on you end, sending for a re-review 💯

**Review by suretrust — APPROVED** [review]
Hi @shubham14p3, 👋 

### Amazing job implementing the suggested changes. 🙌 Let's get this merged! 👍 

_Good luck with your future projects and happy coding!_ 🎉

**Pull request merged**